src/lib/scheduler.ts

```diff
@@ -0,0 +1,51 @@
+import { HttpService, RunService } from "@rbxts/services";
+
+type ScheduledFunction = {
+	name: string | undefined;
+	frequency: number | undefined;
+	func: (deltaTime: number) => void;
+	_lastRan: number | undefined;
+};
+
+const functions: Map<string, ScheduledFunction> = new Map();
+
+RunService.Heartbeat.Connect(() => {
+	functions.forEach((v, k) => {
+		task.spawn(() => {
+			const currentTime = os.clock();
+
+			if (v.frequency !== undefined && v._lastRan !== undefined) {
+				if (currentTime - v._lastRan > 1 / v.frequency) {
+					debug.profilebegin(v.name || `scheduled_${k}`);
+					v.func(currentTime - v._lastRan);
+					debug.profileend();
+
+					v._lastRan = currentTime;
+				}
+			}
```

Comment:
Move this outside the task.spawn, spawn all the necessary tasks after passing this condition a single time globally. Refer to another one of the reviews for more info.

Reply:
Okay